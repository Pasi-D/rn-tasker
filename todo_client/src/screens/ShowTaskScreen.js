import React, { Component } from "react";
import { View } from "react-native";

import { Header, Text, Divider, Button } from "react-native-elements";

import { PropTypes } from "prop-types";

import { connect } from "react-redux";

import Modal from "react-native-modal";

import deleteTask from '../actions/taskActions';

class ShowTaskScreen extends Component{
    state = {
        isDeleteModalVisible: false
    };

    constructor(props) {
        super(props);
    }

    _editButtonPress = (task) => {
        this.props.navigation.push('Edit', {
            task: {...task}
        })
    }

    _toggleDeleteModal = () => {
        this.setState({ 
            isDeleteModalVisible: !this.state.isDeleteModalVisible 
        });
    }

    _confDeleteButton = (id) => {
        this.props.deleteTask(id)
        this._toggleDeleteModal();
    }

    _cancelDeleteButton = () => {
        this._toggleDeleteModal();   
    }


    render(){
        const { navigation } = this.props;
        const task = navigation.getParam('task', {"status":true,"_id":"5b963c959cf78a22a43c0ff2","title":"Task not found","description":"Something went wrong","end_date":"2018-09-21T00:00:00.000Z","start_date":"2018-09-13T00:00:00.000Z","__v":0});
                
        return(
            <View>
                <Header 
                    centerComponent={{ text: 'Task Detail', style: { color: '#fff' } }}
                />
                <Divider style={{ backgroundColor: 'blue' }} />
                <View style={{
                    flex: 1,
                    flexDirection: 'column',
                    justifyContent: 'center',
                    alignItems: 'center',
                    marginTop: 90
                }}>
                    <Text>Task Title : {JSON.stringify(task.title)}</Text>
                    <Text>Task Description : {JSON.stringify(task.description)}</Text>
                    <Text>Start Date: {JSON.stringify(task.start_date)}</Text>
                    <Text>End Date: {JSON.stringify(task.end_date)}</Text>
                </View>
                
                <View style={{marginTop: 70}}>
                    <Button 
                        title='Edit'
                        buttonStyle={{
                            backgroundColor: "rgba(0,255,0,0.3)"
                        }}
                        onPress={() => this._editButtonPress(task)}
                    />
                    <Button
                        title='Delete'
                        buttonStyle={{
                            backgroundColor: "rgba(255,0,0,0.3)"
                        }}
                        onPress={this._toggleDeleteModal}    
                    />
                    <Modal 
                        isVisible={this.state.isDeleteModalVisible}
                        backdropColor="white"
                        backdropOpacity='0.95'
                    >
                        <View style={{ 
                            flex: 1,
                            marginTop: 120
                        }}>
                            <Button
                                title='Confirm'
                                onPress={() => this._confDeleteButton(task._id)}
                                buttonStyle={{
                                    backgroundColor: "rgba(255,0,0,0.3)"
                                }}
                            />
                            <Button 
                                title='Cancel'
                                onPress={() => this._cancelDeleteButton()}
                                buttonStyle={{
                                    backgroundColor: "rgba(0,0,255,0.3)"
                                }}
                            />
                        </View>
                    </Modal>
                </View>                
            </View>
        )
    }

    
}

ShowTaskScreen.propTypes = {
    deleteTask: PropTypes.func.isRequired,
    task: PropTypes.object.isRequired
}

const mapStateToProps = (state) => ({
    tasks: state.tasks
})

export default connect(mapStateToProps, { deleteTask })(ShowTaskScreen);