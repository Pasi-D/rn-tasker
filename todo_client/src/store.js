import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";

//rootReducer
import rootReducers from './reducers';

//rootSaga
import rootSaga from './sagas/sagas'

import createSagaMiddleware from 'redux-saga'

//redux-logger
import { createLogger } from "redux-logger";
const logger = createLogger({
    
})

//create saga middleare
const sagaMiddleware = createSagaMiddleware();

const middleware =  [thunk, sagaMiddleware, logger]

const initialState  = {}

const store = createStore(
    rootReducers,
    initialState,
    compose(
        applyMiddleware(...middleware)
    )
)

/* run the rootSagas file here*/
sagaMiddleware.run(rootSaga);

export default store;