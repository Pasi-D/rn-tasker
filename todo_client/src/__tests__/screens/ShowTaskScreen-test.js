// Use Enzyme along with Jest to Test snapshots 

import "react-native"
import React from "react";
import { shallow } from "enzyme";

import store from '../../store';

import ShowTaskScreen from '../../screens/ShowTaskScreen';

describe('>>> Show Task Screen ', () => {
    it('+++ Should Mount Show Task Screen', () => {
        const props = {
            navigation: {
                getParam() {
                    return (
                        {
                            "status":true,
                            "_id":"5b963c959cf78a22a43c0ff2",
                            "title":"Dummy Task",
                            "description":"This Task is a dummy",
                            "end_date":"2018-09-21T00:00:00.000Z",
                            "start_date":"2018-09-13T00:00:00.000Z",
                            "__v":0
                        }
                    )
                }
            },
            store: store
        }
        
        const wrapper = shallow(<ShowTaskScreen {...props} />)
        expect(wrapper).toMatchSnapshot()

    })
})
