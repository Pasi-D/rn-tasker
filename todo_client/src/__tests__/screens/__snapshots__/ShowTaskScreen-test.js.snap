// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Show Task Screen  +++ Should Mount Show Task Screen 1`] = `
<ShowTaskScreen
  deleteTask={[Function]}
  falsifyAddStatus={[Function]}
  falsifyDelStatus={[Function]}
  navigation={
    Object {
      "getParam": [Function],
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
  tasks={
    Object {
      "added": false,
      "deleted": false,
      "errors": null,
      "loading": true,
      "task": null,
      "tasks": null,
    }
  }
/>
`;
